packages/sandbox/src/helpers/postMessageRelayHelpers.ts

```diff
@@ -247,48 +282,117 @@ export function executeOperation({
         const observable = readMultipartWebStream(response, mimeType);
 
         let isFirst = true;
-        observable.subscribe({
+        const observableSubscription = observable.subscribe({
           next(data) {
-            sendPostMessageToEmbed({
-              message: {
-                // Include the same operation ID in the response message's name
-                // so the Explorer knows which operation it's associated with
-                name: EXPLORER_QUERY_MUTATION_RESPONSE,
-                operationId,
-                response: {
-                  incremental: data.data.incremental,
-                  data: data.data.data,
-                  errors: data.data.errors,
-                  extensions: data.data.extensions,
-                  path: data.data.path,
-                  status: response.status,
-                  headers: isFirst
-                    ? [responseHeaders, ...(data.headers ? [data.headers] : [])]
-                    : data.headers,
-                  hasNext: true,
-                  size: data.size,
+            if (multipartSubscriptionClient) {
+              multipartSubscriptionClient.stopListeningCallback = () => {
+                data.closeReadableStream();
+                observableSubscription.unsubscribe();
+              };
+            }
+            multipartSubscriptionClient?.emit('connected');
```

Comment:
These feel like 1 time setup, but we're doing it everytime we get data. Should we move `multipartSubscriptionClient?.emit('connected');` up to the top of the function when we first get the response (line 270)
And can we return the `closeReadableStream` from `readMultipartWebStream`?